spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -74,17 +100,31 @@
       throw new IllegalArgumentException("Cannot only specify option end-snapshot-id to do incremental scan");
     }
 
-    // look for split behavior overrides in options
-    this.splitSize = Spark3Util.propertyAsLong(options, SparkReadOptions.SPLIT_SIZE, null);
-    this.splitLookback = Spark3Util.propertyAsInt(options, SparkReadOptions.LOOKBACK, null);
-    this.splitOpenFileCost = Spark3Util.propertyAsLong(options, SparkReadOptions.FILE_OPEN_COST, null);
+    this.splitSize = table instanceof BaseMetadataTable ? readConf.metadataSplitSize() : readConf.splitSize();
```

Comment:
What about translating from the metadata property to the normal property in the table definition instead? That avoids breaking the abstraction.

Reply:
Better idea: I think that `TableScan` should report these settings instead of forcing us to pass them in. That way you can configure the scan with the read options if they are present, then get the scan's split settings without needing to forward/rewrite settings through `properties` or go around the API.

That fits with a refactor I want to get to that removes all of the settings from here anyway and creates the `TableScan` in the `ScanBuilder`. There's no need to create a scan in more than one place.

Reply:
Switched to `TableScan`.